packages/lit-element/src/lit-element.ts

```diff
@@ -123,131 +101,20 @@ export class LitElement extends UpdatingElement {
    * it will not needlessly try to `finalize`.
    *
    * Note this property name is a string to prevent breaking Closure JS Compiler
-   * optimizations. See updating-element.ts for more information.
+   * optimizations. See updating-element for more information.
    */
   protected static ['finalized'] = true;
 
-  /**
-   * Array of styles to apply to the element. The styles should be defined
-   * using the [[`css`]] tag function or via constructible stylesheets.
-   */
-  static styles?: CSSResultGroup;
-  private static _elementStyles?: CSSResultFlatArray;
-
-  /**
-   * Options used when calling `attachShadow`. Set this property to customize
-   * the options for the shadowRoot; for example, to create a closed
-   * shadowRoot: `{mode: 'closed'}`.
-   *
-   * Note, these options are used in `createRenderRoot`. If this method
-   * is customized, options should be respected if possible.
-   */
-  static shadowRootOptions: ShadowRootInit = {mode: 'open'};
-
-  /**
-   * Takes the styles the user supplied via the `static styles` property and
-   * returns the array of styles to apply to the element.
-   * Override this method to integrate into a style management system.
-   *
-   * Styles are deduplicated preserving the _last_ instance in the list. This
-   * is a performance optimization to avoid duplicated styles that can occur
-   * especially when composing via subclassing. The last item is kept to try
-   * to preserve the cascade order with the assumption that it's most important
-   * that last added styles override previous styles.
-   *
-   * @nocollapse
-   */
-  protected static finalizeStyles(styles?: CSSResultGroup): CSSResultFlatArray {
-    const elementStyles = [];
-    if (Array.isArray(styles)) {
-      // Dedupe the flattened array in reverse order to preserve the last items.
-      // TODO(sorvell): casting to Array<unknown> works around TS error that
-      // appears to come from trying to flatten a type CSSResultArray.
-      const set = new Set((styles as Array<unknown>).flat(Infinity).reverse());
-      // Then preserve original order by adding the set items in reverse order.
-      for (const s of set) {
-        elementStyles.unshift(getCompatibleStyle(s as CSSResultOrNative));
-      }
-    } else if (styles !== undefined) {
-      elementStyles.push(getCompatibleStyle(styles));
-    }
-    return elementStyles;
-  }
-
-  protected static finalize() {
-    const wasFinalized = super.finalize();
-    if (wasFinalized) {
-      this._elementStyles = this.finalizeStyles(this.styles);
-    }
-    return wasFinalized;
-  }
-
-  /**
-   * Node or ShadowRoot into which element DOM should be rendered. Defaults
-   * to an open shadowRoot.
-   */
-  readonly renderRoot!: HTMLElement | DocumentFragment;
-  readonly _renderOptions!: RenderOptions;
+  readonly _renderOptions: RenderOptions = {eventContext: this};
```

Comment:
Why readonly if we need to modify it? It's private so it's for us.

Reply:
We don't modify `_renderOptions` but we do modify the options themselves. Changed in lit-html.